src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -252,9 +254,15 @@ internal unsafe int IndexOfCore(string source, string target, int startIndex, in
 
             if (options == CompareOptions.Ordinal)
             {
-                index = IndexOfOrdinal(source, target, startIndex, count, ignoreCase: false);
+                index = SpanHelpers.IndexOf(
+                    ref Unsafe.Add(ref source.GetRawStringData(), startIndex),
+                    count,
```

Comment:
should this be `count - startIndex`? could we add assert w.r.t ranges?

Reply:
No, is same as ` pSource + startIndex, count` below under `#endif`

`startIndex` The search starting position
`count` The number of character positions to examine

`count - startIndex` would be too short

Reply:
ah correct, I was thinking of original string length and not the substring